packages/react-components/react-motion-components-preview/library/src/components/Fade/Fade.test.tsx

```diff
@@ -0,0 +1,130 @@
+import * as React from 'react';
+import { Fade, FadeExaggerated, FadeSnappy } from './Fade';
+import { render, waitFor } from '@testing-library/react';
+import { motionTokens } from '@fluentui/react-motion';
+import { mockAnimation } from '../../testing/testUtils';
+
+describe('Fade motion component', () => {
+  let animateSpy: jest.SpyInstance;
+  const testElement = <div data-testid="fade-content">Test</div>;
+
+  beforeAll(() => {
+    HTMLElement.prototype.animate = () => mockAnimation;
+  });
+
+  beforeEach(() => {
+    animateSpy = jest.spyOn(HTMLElement.prototype, 'animate').mockImplementation(() => mockAnimation);
+  });
+
+  afterEach(() => {
+    animateSpy.mockRestore();
+  });
+
+  afterAll(() => {
+    delete (HTMLElement.prototype as unknown as { animate?: Function }).animate;
+  });
+
+  it('should render Fade with correct opacity keyframes, duration and easing (visible=false -> true -> false)', () => {
+    const { rerender } = render(<Fade visible={false}>{testElement}</Fade>);
+
+    // Testing fade in motion
+    rerender(<Fade visible={true}>{testElement}</Fade>);
+    expect(animateSpy).toHaveBeenCalledWith(
+      [{ opacity: 0 }, { opacity: 1 }],
+      expect.objectContaining({ duration: motionTokens.durationNormal, easing: motionTokens.curveEasyEase }),
+    );
+
+    // Testing fade out motion
+    rerender(<Fade visible={false}>{testElement}</Fade>);
+    expect(animateSpy).toHaveBeenCalledWith(
+      [{ opacity: 1 }, { opacity: 0 }],
+      expect.objectContaining({ duration: motionTokens.durationNormal, easing: motionTokens.curveEasyEase }),
+    );
+  });
+
+  it('should render Snappy variant of Fade component with correct opacity keyframes, duration and easing', () => {
+    const { rerender } = render(<FadeSnappy visible={false}>{testElement}</FadeSnappy>);
+
+    rerender(<FadeSnappy visible={true}>{testElement}</FadeSnappy>);
+
+    expect(animateSpy).toHaveBeenCalledWith(
+      [{ opacity: 0 }, { opacity: 1 }],
+      expect.objectContaining({ duration: motionTokens.durationFast, easing: motionTokens.curveEasyEase }),
+    );
+  });
+
+  it('should render Exaggerated variant of Fade component with correct opacity keyframes, duration and easing', () => {
+    const { rerender } = render(<FadeExaggerated visible={false}>{testElement}</FadeExaggerated>);
+
+    rerender(<FadeExaggerated visible={true}>{testElement}</FadeExaggerated>);
+
+    expect(animateSpy).toHaveBeenCalledWith(
+      [{ opacity: 0 }, { opacity: 1 }],
+      expect.objectContaining({ duration: motionTokens.durationGentle, easing: motionTokens.curveEasyEase }),
+    );
+  });
+
+  it('should render child element with overriding styles and still apply Fade animations', () => {
```

Comment:
This test is not able to confirm that the animation overrides the inline style. It only asserts which motion parameters were passed to `animate()`, which doesn't tell us whether the animation will take precedence over the inline style. That could require checking `getComputedStyle`; however, JSDOM in Jest doesn't support the Web Animations API. So it seems this kind of test would need to be done in a real browser, i.e. via Cypress in Fluent v9.